src/gui/widgets/TabWidget.cpp

```diff
@@ -104,27 +119,73 @@ void TabWidget::setActiveTab( int _idx )
 }
 
 
-
-
-void TabWidget::mousePressEvent( QMouseEvent * _me )
+// Return the index of the tab at position "pos"
+int TabWidget::findTabAtPos( const QPoint *pos )
 {
-	if( _me->y() > 1 && _me->y() < 13 )
+
+	if( pos->y() > 1 && pos->y() < m_tabbarHeight - 1 )
 	{
-		int cx = ( ( m_caption == "" ) ? 4 : 14 ) +
-					fontMetrics().width( m_caption );
-		for( widgetStack::iterator it = m_widgets.begin();
-						it != m_widgets.end(); ++it )
+               int cx = ( ( m_caption == "" ) ? 4 : 14 ) + fontMetrics().width( m_caption );
```

Comment:
Here you could replace 14 and 4 with predefined variables, to make changes easier down the road.